File: comparison/results/haiku/baseline/humaneval/HumanEval_1/typescript/solution.ts
function separate_paren_groups(paren_string: string): string[] {
    const cleaned = paren_string.replace(/\s/g, '');
    const result: string[] = [];
    let current = '';
    let depth = 0;
    
    for (const char of cleaned) {
        if (char === '(') {
            depth++;
            current += char;
        } else if (char === ')') {
            current += char;
            depth--;
            if (depth === 0) {
                result.push(current);
                current = '';
            }
        }
    }
    
    return result;
}

File: comparison/results/haiku/baseline/humaneval/HumanEval_1/csharp/solution.cs
public static List<string> SeparateParenGroups(string paren_string) {
    var result = new List<string>();
    string cleaned = paren_string.Replace(" ", "");
    
    string current = "";
    int depth = 0;
    
    foreach (char c in cleaned) {
        if (c == '(') {
            depth++;
            current += c;
        } else if (c == ')') {
            current += c;
            depth--;
            if (depth == 0) {
                result.Add(current);
                current = "";
            }
        }
    }
    
    return result;
}
